hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/kafka/PropertiesResolver.java

```diff
@@ -85,19 +86,27 @@ final class PropertiesResolver {
     private PropertiesResolver() { }
 
     static Properties resolveConsumerProperties(Map<String, String> options) {
```

Comment:
this variant is used only in tests and is strange, because even though it gets `options` it does not extract avro schema from them. Is this method needed?

Reply:
1. Effective schema cannot be extracted from mapping options since it might have been overridden by `__key`/`this` fields.
2. It is legal to resolve producer/consumer properties without providing effective schemas since `PropertiesResolver` is not specific to Avro (and also `UpsertTargetDescriptor.getSchema()` is only implemented for Avro).
3. `PropertiesResolver` is already test-dependent due to `if (schema != null)` condition. We can remove this condition and modify the tests, but would it be worth the effort and ugliness?

Reply:
this variant seems to be a shortcut for `PropertiesResolverTest`, so you can move it there

ad.1. this supports the idea of (re)moving this method as it is incomplete in general
ad.2. yes, but this method is intended to be generic to handle all formats. Having a key/ValueSchame parameter will make the caller think what schema should be passed there.
ad3. That is strange. Is is valid to use HAZELCAST_AVRO_DESERIALIZER without schema? Maybe we should throw exception in such case?

Reply:
this variant seems to be a shortcut for `PropertiesResolverTest`, so you can move it there

ad.1. this supports the idea of (re)moving this method as it is incomplete in general
ad.2. yes, but this method is intended to be generic to handle all formats. Having a key/ValueSchame parameter will make the caller think what schema should be passed there.
ad3. That is strange. Is is valid to use HAZELCAST_AVRO_DESERIALIZER without schema? Maybe we should throw exception in such case?

Reply:
68fc637d